libraries/AR_Motors/AP_MotorsUGV.cpp

```diff
@@ -215,6 +222,86 @@ void AP_MotorsUGV::set_throttle(float throttle)
+    ** (_throttle < 0)  backward mode anyway
+    */
+    if ((_throttle < 0.0f) || ((_throttle <= 0.0f) && (_esc_backwards_unlock.active) )) {
+        const uint32_t now = AP_HAL::millis();
+      
+        switch (_esc_backwards_unlock.state) {
+            case UnlockState::UNLOCK_WAITING:
+                _esc_backwards_unlock.simulated_signal=0;
+                _throttle = 0.0f;
+                _esc_backwards_unlock.last_set_throttle_call = now;
+                _esc_backwards_unlock.active = true;
+                _esc_backwards_unlock.state = UnlockState::UNLOCK_STEP_DEC_TO_MAX_NEG;
+                break;
+
+            case UnlockState::UNLOCK_STEP_DEC_TO_MAX_NEG:
+                if ((now - _esc_backwards_unlock.last_set_throttle_call) < 20) {
+                    break;
+                }
+               
+                _esc_backwards_unlock.simulated_signal-=2.0f;
+                _throttle = _esc_backwards_unlock.simulated_signal;
+                _esc_backwards_unlock.last_set_throttle_call = now;
+
+                if (_esc_backwards_unlock.simulated_signal <= - 98.0f) {
+                    _esc_backwards_unlock.simulated_signal = -100.0f;
+                    _esc_backwards_unlock.state = UnlockState::UNLOCK_STEP_STAY_MAX;
+                }
+                break;
+            
+            case UnlockState::UNLOCK_STEP_STAY_MAX:
+                _throttle = -100.0f;
+
+                if ((now - _esc_backwards_unlock.last_set_throttle_call) > 300) {
+                    _esc_backwards_unlock.state = UnlockState::UNLOCK_STEP_INC_TO_ZERO;
+                    _esc_backwards_unlock.last_set_throttle_call = now;
+                }
+                break;
+            
+            case UnlockState::UNLOCK_STEP_INC_TO_ZERO:
+                if ((now - _esc_backwards_unlock.last_set_throttle_call) < 20) {
+                    break;
+                }
+               
+                _esc_backwards_unlock.simulated_signal+=2.0f;
+                _throttle = _esc_backwards_unlock.simulated_signal;
+                _esc_backwards_unlock.last_set_throttle_call = now;
+
+                if (_esc_backwards_unlock.simulated_signal >= -2.0f) {
+                    _esc_backwards_unlock.simulated_signal = 0.0f;
+                    _esc_backwards_unlock.state = UnlockState::UNLOCK_STEP_ZEROS_2;
+                    _esc_backwards_unlock.last_set_throttle_call = now;
+                }
+                break;
+
+
+            case UnlockState::UNLOCK_STEP_ZEROS_2:
+                _throttle = 0.0f;
+
+                if ((now - _esc_backwards_unlock.last_set_throttle_call) > 50) {
+                    _esc_backwards_unlock.state = UnlockState::UNLOCK_FINISHED;
+                    _esc_backwards_unlock.last_set_throttle_call = now;
+                    _esc_backwards_unlock.active = false;
+                }
+                break;
+
+            case UnlockState::UNLOCK_FINISHED:
+               break;
+        }
+    } else {
```

Comment:
I would be tempted to just do this.
```suggestion
    } elseif (_throttle > 0)) {
```

Reply:
&& (_throttle > 0) is not needed in line 301 as it is used in line 300